ext/standard/streamsfuncs.c

```diff
@@ -756,115 +810,160 @@ static int stream_array_emulate_read_fd_set(zval *stream_array)
 /* }}} */
 
 /* {{{ Runs the select() system call on the sets of streams with a timeout specified by tv_sec and tv_usec */
-PHP_FUNCTION(stream_select)
-{
-	zval *r_array, *w_array, *e_array;
-	struct timeval tv, *tv_p = NULL;
-	fd_set rfds, wfds, efds;
-	php_socket_t max_fd = 0;
-	int retval, sets = 0;
-	zend_long sec, usec = 0;
-	bool secnull;
-	bool usecnull = 1;
-	int set_count, max_set_count = 0;
-
-	ZEND_PARSE_PARAMETERS_START(4, 5)
-		Z_PARAM_ARRAY_EX2(r_array, 1, 1, 0)
-		Z_PARAM_ARRAY_EX2(w_array, 1, 1, 0)
-		Z_PARAM_ARRAY_EX2(e_array, 1, 1, 0)
-		Z_PARAM_LONG_OR_NULL(sec, secnull)
-		Z_PARAM_OPTIONAL
-		Z_PARAM_LONG_OR_NULL(usec, usecnull)
-	ZEND_PARSE_PARAMETERS_END();
+PHP_FUNCTION(stream_select) {
+    zval *r_array, *w_array, *e_array;
+    struct timeval tv, *tv_p = NULL;
+    fd_bigset rfds, wfds, efds;
+    php_socket_t max_fd = 0;
+    int retval, sets = 0;
+    zend_long sec, usec = 0;
+    bool secnull;
+    bool usecnull = 1;
+    int set_count = 0, max_set_count = 0;
+
+    ZEND_PARSE_PARAMETERS_START(4, 5)
+        Z_PARAM_ARRAY_EX2(r_array, 1, 1, 0)
+        Z_PARAM_ARRAY_EX2(w_array, 1, 1, 0)
+        Z_PARAM_ARRAY_EX2(e_array, 1, 1, 0)
+        Z_PARAM_LONG_OR_NULL(sec, secnull)
+        Z_PARAM_OPTIONAL
+        Z_PARAM_LONG_OR_NULL(usec, usecnull)
+    ZEND_PARSE_PARAMETERS_END();
 
-	FD_ZERO(&rfds);
-	FD_ZERO(&wfds);
-	FD_ZERO(&efds);
+#ifdef PHP_WIN32
+	/* Check if the Windows standard handle is redirected to file */
+	long max_fds = FD_SETSIZE;
+#else
+	/* Check if the file descriptor identifier is a terminal */
+	long max_fds = sysconf(_SC_OPEN_MAX);
+	if (max_fds == -1) {
+		max_fds = FD_SETSIZE;
```

Comment:
I suggest to initialize to a large size here

Reply:
I set it to 131072 now. Thanks!